lib/deployer.rb

```diff
@@ -0,0 +1,42 @@
+require "rake/file_utils"
+
+class Deployer
+  include FileUtils
+
+  attr_reader :build_dir, :deploy_branch, :deploy_sha
+
+  def self.prepare_deploy(options={})
+    self.new(options).prepare_deploy
+  end
+
+  def initialize(options={})
+    @build_dir = options.fetch(:build_dir, "_site")
+    @deploy_branch = options.fetch(:deploy_branch, "gh-pages")
+    @deploy_sha = options.fetch(:deploy_sha)
+  end
+
+  def prepare_deploy
+    build
+    prepare_deploy_branch
+    promote_build
+  end
```

Comment:
Yeah, it makes sense to have some messaging around what's happening. I'll play around with it, but at the very least a message at the end would be nice.

We can remove the messaging when we move to auto deploys.